packages/medusa/src/services/event-bus.ts

```diff
@@ -225,23 +239,35 @@ export default class EventBusService {
   async emit<T>(
     eventName: string,
     data: T,
-    options: Record<string, unknown> & EmitOptions = { attempts: 1 }
-  ): Promise<StagedJob | void> {
-    const opts: EmitOptions = {
-      removeOnComplete: {
-        age: COMPLETED_JOB_TTL,
-      },
-      ...options,
-    }
-
-    if (typeof options.attempts === "number") {
-      opts.attempts = options.attempts
-      if (isDefined(options.backoff)) {
-        opts.backoff = options.backoff
+    options?: Record<string, unknown> & EmitOptions
+  ): Promise<StagedJob | void>
+
+  async emit<
+    T,
+    TInput extends string | EmitData<T>[] = string,
+    TResult = TInput extends EmitData<T>[] ? StagedJob[] : StagedJob
+  >(
+    eventNameOrData: TInput,
+    data?: T,
+    options?: Record<string, unknown> & EmitOptions
+  ): Promise<TResult | void> {
+    let events = eventNameOrData as EmitData<T>[]
+
+    if (isString(eventNameOrData)) {
```

Comment:
suggestion(non-blocking): Extract to const `const isBulkEmit = !isString(eventNameOrData)` as we are using it two places in the method